runtime/main.cc

```diff
@@ -0,0 +1,29 @@
+//===-- runtime/main.cc -----------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//----------------------------------------------------------------------------//
+
+#include "main.h"
+#include "terminator.h"
+#include <cfenv>
+#include <cstdlib>
+
+extern "C" {
+
+void __FortranProgram();  // PROGRAM statement
+
+int main(int argc, const char *argv[], const char *envp[]) {
+  Fortran::runtime::argc = argc;
+  Fortran::runtime::argv = argv;
+  Fortran::runtime::envp = envp;
+  std::feclearexcept(FE_ALL_EXCEPT);
+  std::fesetround(FE_TONEAREST);
+  std::atexit(Fortran::runtime::NotifyOtherImagesOfNormalEnd);
+  // TODO: Runtime configuration settings from environment
+  __FortranProgram();
+  return EXIT_SUCCESS;
```

Comment:
Technically this return should never be hit because `ProgramEndStatement` should be called by `__FortranProgram` right ?

Reply:
It might be more convenient to have the Fortran program return to `main`, I think.

Reply:
Is a user PROGRAM main, SUBROUTINE main, bind(C) main, or other use of "main" mangled or otherwise accounted for to avoid conflict?

Reply:
> Is a user PROGRAM main, SUBROUTINE main, bind(C) main, or other use of "main" mangled or otherwise accounted for to avoid conflict?

Yes, yes, no, and yes, I think.